cmake/FindAVX.cmake

```diff
@@ -1,64 +1,58 @@
 cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
 
-# Code adapted from PyTorch:
```

Comment:
I noticed that this code was returning false positives (i.e., detecting that we can use `-mavx512f` flag on AMD when we shouldn't). It was also really cumbersome to read through. I refactored it based on this: https://raw.githubusercontent.com/hideo55/CMake-FindSSE/master/FindSSE.cmake


This also adds SSE support on MacOS.